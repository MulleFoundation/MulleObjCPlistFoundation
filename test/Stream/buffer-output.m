#import <MulleObjCPlistFoundation/MulleObjCPlistFoundation.h>


@interface SimpleStream : NSObject <MulleObjCOutputStream>
@end


@implementation SimpleStream

- (void) mulleWriteBytes:(void *) bytes
                  length:(NSUInteger) length
{
   printf( "%.*s", (int) length, (char *) bytes);
}

@end


int   main( void)
{
   MulleObjCBufferedOutputStream   *stream;
   SimpleStream                    *simple;
   int                             i;
   char                            buf[ 64];

   simple = [SimpleStream object];
   stream = [[[MulleObjCBufferedOutputStream alloc] initWithOutputStream:simple
                                                              bufferSize:32] autorelease];
   for( i = 1848-15; i <= 1848; i++)
   {
      sprintf( buf, "VfL Bochum %d\n", i);
      [stream mulleWriteBytes:buf
                       length:-1];
   }
   [stream flush];

   return( 0);
}
