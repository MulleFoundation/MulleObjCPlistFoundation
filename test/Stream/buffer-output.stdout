VfL Bochum 1833
VfL Bochum 1834
VfL Bochum 1835
VfL Bochum 1836
VfL Bochum 1837
VfL Bochum 1838
VfL Bochum 1839
VfL Bochum 1840
VfL Bochum 1841
VfL Bochum 1842
VfL Bochum 1843
VfL Bochum 1844
VfL Bochum 1845
VfL Bochum 1846
VfL Bochum 1847
VfL Bochum 1848
